File: Data Structures & Algorithms/sliding-window-maximum/submission-2.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        deque<pair<int,int>> series;
        for (int i=0;i<k;i++){
            if(series.empty()){
                series.push_back(make_pair(nums[i],i));
            }
            else{
                while(!series.empty() and series.back().first<nums[i]){
                    series.pop_back();
                }
                series.push_back(make_pair(nums[i],i));
            }
        }
        vector<int> ans;
        ans.push_back(series.front().first);
        int indx = k;
        while(indx<nums.size()){
            int removeIdx = indx-k;
            // cout
            if(series.front().second<=removeIdx){
                series.pop_front();
            }
            while(!series.empty() and (series.back().first<nums[indx])){
                    series.pop_back();
                }
            series.push_back(make_pair(nums[indx],indx));
            ans.push_back(series.front().first);
            indx++;
        }
        return ans;

    }
};

File: Data Structures & Algorithms/generate-parentheses/submission-2.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        ans = []
        def req(i,j,st):

            if i==0 and j==0:
                ans.append(st)
                return
            
            if i==j:
                req(i-1,j,st+"(")
            
            else:
                if i>0:
                    req(i-1,j,st+"(")
                req(i,j-1,st+")")
        i = n
        j = n
        req(i,j,"")
        return ans

